opensearch/sycamore-opensearch.sh

```diff
@@ -423,9 +386,136 @@ deploy_model() {
     if [[ -z "${model_id}" ]]; then
         wait_task "${task_id}" "$name model to become ready"
         model_id="${GET_TASK_MODEL_ID}"
-        echo "Fetched $name model id $model_id for task_id $task_id"
+        echo "Fetched $name model id $model_id for task_id $task_id" >&2
+    else
+        echo "Using existing model id ${model_id}" >&2
+    fi
+
+    # Create deploy task
+    local deploy_model_log_file="${ARYN_STATUSDIR}/curl.better_deploy_model_task.${name}"
+    spawn_deploy_model_task "${name}" "${model_id}"
+    local deploy_task_id="$(jq -r '.task_id' "${deploy_model_log_file}")"
+
+    # Cycle on the task status
+    debug "Wait for deploy task to finish"
+    local deploy_status_file="${ARYN_STATUSDIR}/curl.deploy_task_status.${name}"
+    local deploy_task_search_file="${ARYN_STATUSDIR}/curl.deploy_task_search_result.${name}"
+    local i
+    local max_reps=60
+    for i in $(seq "${max_reps}"); do
+        _curl "${BASE_URL}/_plugins/_ml/tasks/${deploy_task_id}" -o "${deploy_status_file}"
+        local status="$(jq -r '.state' "${deploy_status_file}")"
+        debug "${status}"
+        # Case 1: RUNNING / state not found. Task is still running so wait.
+        if [[ "${status}" == 'null' || "${status}" == 'RUNNING' || "${status}" == 'CREATED' ]]; then
+            echo "Waiting for ${name} to deploy... ${i}/${max_reps}" >&2
+            sleep 1
+            continue
+        # Case 2: COMPLETED. Task is completed, so exit
+        elif [[ "${status}" == 'COMPLETED' ]]; then
+            echo "Deployed ${name} successfully" >&2
+            MODEL_ID="${model_id}"
+            return 0
+        # Case 3: FAILED. Handle some error cases, fail in unrecognized ones.
+        elif [[ "${status}" == 'FAILED' ]]; then
+            handle_deploy_error "${model_id}" "${name}" && return 0
```

Comment:
I think you want `&& return 1` Near as I can tell when handle_deploy_error fails it's an error.

Reply:
the logic I was going for was: when handle_deploy_error succeeds, the model is deployed already. When it's an unrecoverable error, handle_deploy_error dies. When handle_deploy_error fails, then it's a handled error and we can try again